source/technical/complex.rst

```diff
@@ -739,9 +739,52 @@ The ``SoftwareDeploymentGroup`` resources ``export_hosts_sdg`` and ``populate_ho
 
 Finally, the instance ``ResourceGroup`` is configured so that each instance uses the following contextualization method instead of a ``user_data`` script:
 
-.. code:: 
+.. code::
 
    user_data_format: SOFTWARE_CONFIG
    software_config_transport: POLL_SERVER_HEAT
 
 You can follow the same template pattern to configure your own deployment requiring all-to-all information exchange.
+
+.. _advanced-reservation-orchestration:
+
+Advanced Reservation Orchestration
+_______________________________
+
+On Chameleon you can configure a Heat Stack to launch as soon as your lease begins. Whether your experiments require a large cluster or a single node, orchestrating an advanced reservation is can save you time configuring your environment or provide a blueprint for your experiment that will run automatically
+when the necessary resources become available.
+
+At present, you will need to use our customized versions the Heat and Blazar CLI tools to implement this feature. We are currently working to provide support for this functionality through the GUI.
+
+Install Custom CLI
+~~~~~~~~~~~
+
+Clone the Chameleon repositories for ``python-heatclient`` and ``python-blazarclient``. You can then simply run ``python setup.py install`` in the repo folder to override the your existing python client packages if they are already installed. Links and sample code below:
+
+.. code::
+
+    git clone https://github.com/ChameleonCloud/python-heatclient.git
```

Comment:
People may find it easier to install via `pip`:

```
pip install git+git://github.com/ChameleonCloud/python-heatclient.git
pip install git+git://github.com/ChameleonCloud/python-blazarclient.git
```

Reply:
Yep, that's simpler. Didn't know you could do that.

Reply:
Even better is to use `pip install git+https://`, in case people cannot access the Git port due to restrictive firewall policies.